DQMOffline/Trigger/src/EgHLTMonElemFuncs.cc

```diff
@@ -95,6 +96,99 @@ void MonElemFuncs::initStdPhoHists(std::vector<MonElemManagerBase<OffPho>*>& his
 							     bins.etaVsPhi.nrY,bins.etaVsPhi.yMin,bins.etaVsPhi.yMax,
 							     &OffPho::hltEta,&OffPho::hltPhi));
 }
+
+
+void MonElemFuncs::initStdEleHistsHEP(std::vector<MonElemManagerBase<OffEle>*>& histVec,const std::string& filterName,const std::string& baseName,const BinData& bins)
```

Comment:
so for all the fancy templating of this code - the design falls apart here.  all the if (hep).. can go away, it would seem, if MonElemFuncs had a derived class for hep17. Was that considered? All the ifs make the details of the monitoring code completely un-understandable imho.